### ISSUE TITLE
Add adjust return type assist

### ISSUE THREAD

**Issue by kpodsiad** [rust-lang/rust-analyzer#14616]
Hey folks, what do you think about adding above-mentioned assist? It could be useful when doing some refactors. If this makes sense, I can try to do it as this will boost my dev workflow. If not, feel free to close this issue.

silly example: If I change `Ok(Foo)` to `Ok(Bar)` I have to adjust return type by myself, but I'd like to quick fix it, if possible. I browsed through already existing assists and there is `add_return_type` but I can't see anything which suits my case.
```rust
struct Foo;
struct Bar;
fn foo() -> Foo {
  Foo // -> change to Bar
}
```

```rust
struct Foo;
struct Bar;
fn foo() -> Foo {
  Bar // offer quick fix action
}
```

**Comment by Veykril** [comment]
This should be a diagnostic quickfix for type mismatches in return position I think, one question i'd have here is if you have something like
```rs
fn foo(bar: bool) -> i32 {
	if bar { return 0.0; }
	0
}

```
This should emit a diagnostic at the end of the function, should the quick fix be applicable here? Using it would introduce a type mismatch somewhere else, but I think that'd be probably fine?

The place to do this would be here https://github.com/rust-lang/rust-analyzer/blob/master/crates/ide-diagnostics/src/handlers/type_mismatch.rs but you might need to thread more info to the diagnostic (as in where this occured/in what context, so you know that the mismatch was due to the return type differing)

**Comment by HKalbasi** [comment]
It is also applicable for other places with explicit type, like a `let x: Ty = exp;`. Also a similar assist can change the expression instead of the type.
